src/surfing_penguin/db_interface.py

```diff
@@ -60,15 +59,14 @@ def get_all_questions(name):
         s = session.query(Survey).filter_by(surveyname=name).first()
         return s.questions
 
-    def new_question(self, data):
+    def new_question(survey, data):
         # data should be a dictionary, with key "content" and "title"
         question = Question()
         question.title = data["title"]
         question.content = data["content"]
-        self.iter = self.iter + 1
-        question.idx = self.iter
-        s = session.query(Survey).filter_by(surveyname=self.name).first()
-        question.survey_id = s.id
+        survey.question_num += 1
+        question.idx = survey.question_num
+        question.survey_id = survey.id
```

Comment:
I mean it should be in \_\_init\_\_() or set_question() of the __Question__ class, not SurveyFunctions. Setting Question's attributes one by one in new_question() is a bit redundant. It's not wrong but it could be simpler for readability.